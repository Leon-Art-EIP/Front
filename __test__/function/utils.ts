import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
