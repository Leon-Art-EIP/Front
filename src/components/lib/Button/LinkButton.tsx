import { ElementType } from "react";
import Button, { IButtonProps } from "./Button";

export interface ILinkButtonProps extends IButtonProps {
  href: string;
  link: ElementType<{ children: JSX.Element; href: string }>;
}

export default function LinkButton({ link: Link, ...props }: ILinkButtonProps): JSX.Element {
  return (
    <Link href={props.href} className={props.className}>
      <Button type={props.type} color={props.color} onClick={props.onClick} loading={props.loading}>
        {props.children}
      </Button>
    </Link>
  );
}
