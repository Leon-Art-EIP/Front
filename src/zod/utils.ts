export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
