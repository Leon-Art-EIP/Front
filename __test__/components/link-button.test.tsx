import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import LinkButton from "../../src/components/lib/Button/LinkButton";

// Composant de lien simulÃ© pour les tests
const MockLink = ({ children, href }: { children: JSX.Element; href: string }) => (
  <a href={href} className="mock-link">
    {children}
  </a>
);

test("LinkButton primary snapshot", () => {
  testSnapshot(
    <LinkButton link={MockLink} href="/test" type="button" color="primary" width="full">
      Primary Button
    </LinkButton>
  );
});

test("LinkButton secondary snapshot", () => {
  testSnapshot(
    <LinkButton link={MockLink} href="/test" type="button" color="secondary">
      Secondary Button
    </LinkButton>
  );
});

test("LinkButton with loading state snapshot", () => {
  testSnapshot(
    <LinkButton link={MockLink} href="/loading" type="submit" color="info" loading={true}>
      Loading Button
    </LinkButton>
  );
});

test("LinkButton danger snapshot", () => {
  testSnapshot(
    <LinkButton link={MockLink} href="/danger" type="reset" color="danger">
      Danger Button
    </LinkButton>
  );
});
