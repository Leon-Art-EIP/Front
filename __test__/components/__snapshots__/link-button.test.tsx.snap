// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkButton danger snapshot 1`] = `
<a
  className="mock-link"
  href="/danger"
>
  <button
    className="rounded-3xl px-16 py-3 font-semibold text-base transition duration-200 flex items-center justify-center bg-primary text-white hover:bg-primary-hover disabled:bg-primary-disabled w-auto"
    type="reset"
  >
    Danger Button
  </button>
</a>
`;

exports[`LinkButton primary snapshot 1`] = `
<a
  className="mock-link"
  href="/test"
>
  <button
    className="rounded-3xl px-16 py-3 font-semibold text-base transition duration-200 flex items-center justify-center text-white bg-primary hover:bg-primary-hover disabled:bg-primary-disabled w-full"
    type="button"
  >
    Primary Button
  </button>
</a>
`;

exports[`LinkButton secondary snapshot 1`] = `
<a
  className="mock-link"
  href="/test"
>
  <button
    className="rounded-3xl px-16 py-3 font-semibold text-base transition duration-200 flex items-center justify-center text-tertiary bg-secondary hover:bg-secondary-hover disabled:bg-secondary-disabled disabled: text-secondary w-auto"
    type="button"
  >
    Secondary Button
  </button>
</a>
`;

exports[`LinkButton with loading state snapshot 1`] = `
<a
  className="mock-link"
  href="/loading"
>
  <button
    className="rounded-3xl px-16 py-3 font-semibold text-base transition duration-200 flex items-center justify-center bg-purple-800 text-white hover:bg-purple-900 disabled:bg-purple-400 w-auto"
    disabled={true}
    type="submit"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary bg-purple-800 text-white hover:bg-purple-900 disabled:bg-purple-400 css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20}
          strokeWidth={4}
          style={{}}
        />
      </svg>
    </span>
  </button>
</a>
`;
